static/app/views/starfish/views/spanSummary/index.tsx

```diff
@@ -146,29 +133,84 @@ export default function SpanSummary({location, params}: Props) {
     initialData: [],
   });
 
-  const spanSamplesQuery = getSpanSamplesQuery({
+  const commonSamplesQueryOptions = {
     groupId,
     transactionName,
     user,
     datetime: pageFilter.selection.datetime,
-    sortBy: state.selectedOption.value,
     p50,
-  });
+  };
 
-  const {isLoading: areSpanSamplesLoading, data: spanSampleData} = useQuery({
+  const commonQueryOptions = {
     queryKey: [
-      'spanSamples',
       groupId,
       transactionName,
       user,
       pageFilter.selection.datetime,
-      state.selectedOption,
+      'spanSamples',
     ],
-    queryFn: () => fetch(`${HOST}/?query=${spanSamplesQuery}`).then(res => res.json()),
     retry: false,
     initialData: [],
+  };
+
+  const results = useQueries({
+    queries: [
+      {
+        ...commonQueryOptions,
+        queryKey: [...commonQueryOptions.queryKey, 'spanSamplesSlowest'],
+        queryFn: () =>
+          fetch(
+            `${HOST}/?query=${getSpanSamplesQuery({
+              ...commonSamplesQueryOptions,
+              populationType: SamplePopulationType.SLOWEST,
+            })}`
+          ).then(res => res.json()),
+      },
+      {
+        ...commonQueryOptions,
+        queryKey: [...commonQueryOptions.queryKey, 'spanSamplesMedian'],
+        queryFn: () =>
+          fetch(
+            `${HOST}/?query=${getSpanSamplesQuery({
+              ...commonSamplesQueryOptions,
+              populationType: SamplePopulationType.MEDIAN,
+            })}`
+          ).then(res => res.json()),
+      },
+      {
+        ...commonQueryOptions,
+        queryKey: [...commonQueryOptions.queryKey, 'spanSamplesFastest'],
+        queryFn: () =>
+          fetch(
+            `${HOST}/?query=${getSpanSamplesQuery({
+              ...commonSamplesQueryOptions,
+              populationType: SamplePopulationType.FASTEST,
+            })}`
+          ).then(res => res.json()),
+      },
+    ],
   });
 
+  const {isLoading: areSpanSamplesLoading, data: spanSampleData} = results.reduce(
+    (acc: {data: any[]; isLoading: boolean; spanIds: Set<string>}, result) => {
+      if (result.isLoading) {
+        acc.isLoading = true;
+        return acc;
+      }
+
+      // Ensures that the same span is not added twice, since there could be overlap in the case of sparse data
```

Comment:
Here's an idea to deal with this better/more consistently:
I think you're already doing a query to find the overall p50, I say, we also get the p75 and p25 (we'll need to add this as a metric for the known needed data if we go this route)

then you do the 3 other queries:
slow: select ... from table where exclusive_time > p75, order by desc exclusive_time
median: select ... from table where exclusive_time < p75 and exclusive_time > p25, order by desc exclusive_time
fast: select ... from table where exclusive_time < p25, order by desc exclusive_time

this way, you can also do consistent pagination. lmk what you think. doesn't have to be addressed in this PR.

Reply:
@shruthilayaj Hmm I like this idea, but I'm not sure if it's worth it yet to add more data to support only this usecase. You do bring up a good point about pagination though, maybe we can look into this later?